### PR TITLE
Fix NF trying to run when old version of nextflow

### PR DESCRIPTION
**Pull request by jfy133**
Now stops with understandable error

Adresses: https://github.com/nf-core/eager/issues/344

Thanks @hughcross for originally 'discovering this' and @pditommaso for pointing us towards the correct fix.

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] `CHANGELOG.md` is updated

**Learn more about contributing:** https://github.com/nf-core/eager/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by drpatelh** [comment]
Hi @jfy133 . This is something we should probably include in the template too. In fact, the linting is failing because it isnt:
https://github.com/nf-core/tools/blob/9e33e7f02758964a1a09436af1eb6afe0e596e4c/nf_core/lint.py#L467

Not entirely sure there is a proper way around this :thinking: until it's added in the next release. You could lint with the `dev` version of `tools` like @MaxUlysse but that isn't ideal.

**Comment by jfy133** [comment]
> Hi @jfy133 . This is something we should probably include in the template too. In fact, the linting is failing because it isnt:
> https://github.com/nf-core/tools/blob/9e33e7f02758964a1a09436af1eb6afe0e596e4c/nf_core/lint.py#L467
> 
> Not entirely sure there is a proper way around this  until it's added in the next release. You could lint with the `dev` version of `tools` like @MaxUlysse but that isn't ideal.

Ha, I just asked in slack about this. I could instead take the approach suggest to me on gitter which is to make my 'own' check in the condition...

**Comment by MaxUlysse** [comment]
For the dev linting, I'm using:
https://github.com/MaxUlysse/sarek/blob/e7852b30f712a1c6ce135eda716f6f9d5cd5eab4/.github/workflows/linting.yml#L50

File: main.nf [hunk 1]
```diff
@@ -235,6 +235,13 @@ where_are_my_files = file("$baseDir/assets/where_are_my_files.txt")
 * SANITY CHECKING
 */
 
+// NF version check! Will replace with manifest system once allwoed by nf-core/tools
```
Comment (MaxUlysse) [inline main.nf]:
```suggestion
// NF version check! Will replace with manifest system once allowed by nf-core/tools
```

**Pull request merged**